lgn-provers/src/provers/mod.rs

```diff
@@ -1,21 +1,17 @@
 use lgn_messages::types::MessageEnvelope;
 use lgn_messages::types::MessageReplyEnvelope;
+use lgn_messages::types::ReplyType;
+use lgn_messages::types::TaskType;
 
 pub mod v1;
 
-/// The prover trait that accepts [`MessageEnvelope`] and is able to process tasks of type
-/// [`TaskType`].
-pub trait LgnProver<T, R> {
-    /// Run the prover with the given [`MessageEnvelope`] and return the result as a
-    /// [`MessageReplyEnvelope`].
-    ///
-    /// # Arguments
-    /// * `envelope` - The [`MessageEnvelope`] that contains the task to be processed.
-    ///
-    /// # Returns
-    /// The result of processing the task as a [`MessageReplyEnvelope`].
+/// Prover trait.
+///
+/// Implementors of this trait shall support some of the message types, for each
+/// support message types it must generate a proof.
+pub trait LgnProver {
     fn run(
         &self,
-        envelope: &MessageEnvelope<T>,
-    ) -> anyhow::Result<MessageReplyEnvelope<R>>;
+        envelope: &MessageEnvelope<TaskType>,
+    ) -> anyhow::Result<MessageReplyEnvelope<ReplyType>>;
 }
```

Comment:
`LgnProver` is only used with `TaskType` and `ReplyType`, remove the generics to simplify the code (this simplifies the trait bounds in the rest of the code).